### PR TITLE
Add namespace parameter in legacy config

### PR DESCRIPTION
**Pull request by chouetz**
Hello
I propose to add the namespace in field of the LegacyGalaxyInfo class to solve the following:

I'm trying to validate a legacy role with both `galaxy-importer.main` and `ansible-lint`
If I have NO namespace in my `meta/main.yaml` I have the following error when executing ansible-lint:
```
ERROR    Computed fully qualified role name of datadog does not follow current galaxy requirements.
Please edit meta/main.yml and assure we can correctly determine full role name:

galaxy_info:
role_name: my_name  # if absent directory name hosting role is used instead
namespace: my_galaxy_namespace  # if absent, author is used instead

Namespace: https://galaxy.ansible.com/docs/contributing/namespaces.html#galaxy-namespace-limitations
Role: https://galaxy.ansible.com/docs/contributing/creating_role.html#role-names

As an alternative, you can add 'role-name' to either skip_list or warn_list.

Computed fully qualified role name of datadog does not follow current galaxy requirements.
Please edit meta/main.yml and assure we can correctly determine full role name:

galaxy_info:
role_name: my_name  # if absent directory name hosting role is used instead
namespace: my_galaxy_namespace  # if absent, author is used instead

Namespace: https://galaxy.ansible.com/docs/contributing/namespaces.html#galaxy-namespace-limitations
Role: https://galaxy.ansible.com/docs/contributing/creating_role.html#role-names

As an alternative, you can add 'role-name' to either skip_list or warn_list.
```
and when I have a namespace and execute galaxy-importer:
```
ERROR: The import failed for the following reason: unknown field in galaxy_info
```

Tests performed with 
```
Importing with galaxy-importer 0.4.13
ansible-lint 6.14.3 using ansible 2.15.4
```

With this PR I allow having a namespace in the configuration but still require namespace to be provided for galaxy-importer (`python -m galaxy_importer.main --legacy-role my-role --namespace my-namespace`)

### CODE REVIEW

**Comment by chouetz** [comment]
@jctanner sorry to bother you again, would this proposal make sense as well?
Many thanks in advance

**Comment by jctanner** [comment]
@chouetz legacy namespaces in galaxy_ng aren't actually restricted to the same set of characters as the v3/collection namespaces. We have to allow any valid github username to become a legacy namespace to assert compatibility with all the roles currently on galaxy.ansible.com

**Comment by chouetz** [comment]
@jctanner I updated the regex. According to what I saw on galaxy.ansible.com we have to enable underscore whereas it's not allowed for github username, I guess.
Many thanks for your feedback

**Comment by utoddl** [comment]
The line above this change, Line 61, has a bad regexp also.
```py
LEGACY_ROLE_NAME_REGEXP = re.compile("^[a-zA-Z0-9-_]{1,55}$")
```
should be
```py
LEGACY_ROLE_NAME_REGEXP = re.compile("^[a-zA-Z0-9_-]{1,55}$")
```
As it is, it matches `'9' through '_'`, not `underscore or hyphen`.

**Comment by chouetz** [comment]
@utoddl @jctanner could you please review the PR please?
Many thanks in advance

**Review by utoddl — APPROVED** [review]

**Comment by utoddl** [comment]
I approved, @chouetz , but I'm only "grey check" worthy. (I don't have write access.) You'll need an approval from reviewer with "green check" super powers. :>

**Comment by jctanner** [comment]
I've dumped the role namespaces and names from old-galaxy to check these regexes ...

https://gist.github.com/jctanner/a3573e1583897bd24408c2ac3d115032/edit

Some are expected failures such as ...

- FAILED - id:15553 namespace:/etc/ansible/roles/tt-test.nfs
- FAILED - id:11063 namespace:kasunmaduraeng@gmail.com

However, I believe we need to allow roles like these ...

- FAILED - id:3487 namespace:george.shuklin
- FAILED - id:23881 namespace:pilou-

```
(venv) [jtanner@p1 galaxy_ng.legacy_role_mapping]$ ansible-galaxy role search -s https://old-galaxy.ansible.com pilou-

Found 1 roles matching your search:

 Name              Description
 ----              -----------
 pilou-.postgresql Install and configure PostgreSQL, dependencies, extensions, databases and users.
```

```
(venv) [jtanner@p1 galaxy_ng.legacy_role_mapping]$ ansible-galaxy role search -s https://old-galaxy.ansible.com george.shuklin

Found 4 roles matching your search:

 Name                                      Description
 ----                                      -----------
 george.shuklin.flow-tools                 Install and configure flow-capture from flow-tools package
 george.shuklin.get-external-ip-via-dyndns Set fact about machine - external (white) IPv4 address via opendns server.
 george.shuklin.inventory_to_hostname      Set up hostname according to inventory name
 george.shuklin.ovs-bridges                Create ovs bridges and add interface to it
```

```
(venv) [jtanner@p1 galaxy_ng.legacy_role_mapping]$ ansible-galaxy role install -p /tmp/test.roles -s https://old-galaxy.ansible.com george.shuklin.inventory_to_hostname
Starting galaxy role install process
- downloading role 'inventory_to_hostname', owned by george.shuklin
- downloading role from https://github.com/amarao/ansible-inventory_to_hostname/archive/master.tar.gz
- extracting george.shuklin.inventory_to_hostname to /tmp/test.roles/george.shuklin.inventory_to_hostname
- george.shuklin.inventory_to_hostname (master) was installed successfully
```

File: galaxy_importer/constants.py
Comment (jctanner) [inline galaxy_importer/constants.py:67]: This neesd to include namespace names that end with `-` and contain `.`

**Review by jctanner — CHANGES_REQUESTED** [review]

File: galaxy_importer/constants.py
Comment (jctanner) [reply galaxy_importer/constants.py:67]: @chouetz I think this has become a critical issues. I suspect you have a busy schedule and can't devote your attention to this issue, so I'm going to give it till the end of the day for you to address my review comments. If you are unable to, I'll try to push fixes to your branch or create a new PR with your commits cherry picked. Thanks for the help on this one, it's greatly appreciated.

Comment (chouetz) [reply galaxy_importer/constants.py]: I've just updated the PR right now, here is the output of your gist with the updated regex
```
(venv) {15:54:19}nicolas.schweitzer@mac~/Downloads/galaxy ➭ python check_roles_regex.py
FAILED - id:6949 namespace:_Harsha_
FAILED - id:11063 namespace:kasunmaduraeng@gmail.com
FAILED - id:14822 namespace:mhabrnal@redhat.com
FAILED - id:14826 namespace:–role-name
FAILED - id:15553 namespace:/etc/ansible/roles/tt-test.nfs
FAILED - id:16048 namespace:rohitggarg/docker-swarm#
FAILED - id:16050 namespace:docker-swarm/tree/master/roles
FAILED - id:18892 namespace:childbdm@gmail.com
FAILED - id:21867 namespace:shashikanth.komandoor@gmail.com
FAILED - id:21913 namespace:ggriffin924@gmail.com
FAILED - id:22855 namespace:piotrfaj@gmail.com
FAILED - id:23153 namespace:farandir@gmail.com
FAILED - id:23178 namespace:lifeblood@163.com
FAILED - id:24233 namespace:ajayshankeryadav@yahoo.co.in
FAILED - id:25461 namespace:ch680351034@gmail.com
FAILED - id:26481 namespace:mheiges@gmail.com
```
I've just allowed names ending with underscores on top of your requirement

Comment (jctanner) [reply galaxy_importer/constants.py]: ty!

**Pull request merged**
